server/src/main/java/org/elasticsearch/index/search/stats/SearchStats.java

```diff
@@ -117,6 +117,22 @@ public void add(Stats stats) {
             suggestCurrent += stats.suggestCurrent;
         }
 
+        public void addForClosedShard(Stats stats) {
+            queryCount += stats.queryCount;
+            queryTimeInMillis += stats.queryTimeInMillis;
+
+            fetchCount += stats.fetchCount;
+            fetchTimeInMillis += stats.fetchTimeInMillis;
+
+            scrollCount += stats.scrollCount;
+            scrollTimeInMillis += stats.scrollTimeInMillis;
+            // need consider the count of the shard's current scroll
+            scrollCount += stats.scrollCurrent;
```

Comment:
I agree that we should not add to the current count since we won't be able to decrement this counter when scroll finishes after the close. However I wonder if we should just ignore the current scrolls for the global `scrollCount`. We won't add `scrollTimeInMillis` for the current scrolls so it would be more accurate to ignore them in the global count ?

Reply:
`scrollCount` should add current scrolls ,this metric means how many `scroll ` action against our index  weather the `scroll` action success or fails. 

but `scrollTimeInMillis` indeed `inaccurate ` for closed or deleted index, ignore `scrollTimeInMillis`  for current scrolls should be suitable.

Reply:
fine with me